### ISSUE TITLE
Simplify allocate statement in generated Fortran code

### ISSUE THREAD

**Issue by Beliavsky** [pyccel/pyccel#608]
pyccel on 

```
x = [3,5,7]
print(x)
```
gives

```
program prog_xlist
use ISO_C_BINDING
implicit none
integer(C_LONG_LONG), allocatable :: x(:)
allocate(x(0:2_C_LONG_LONG - 1_C_LONG_LONG))
x = [3_C_LONG_LONG, 4_C_LONG_LONG]
print *, x
end program prog_xlist
```
Preferably pyccel would do the subtraction, and the allocate statement would be

`allocate(x(0:1_C_LONG_LONG))`

**Comment by EmilyBourne** [comment]
This is a duplicate of issue #564 which contains the implementation details

**Comment by EmilyBourne** [comment]
Duplicate of #564 which contains the implementation details

**Comment by EmilyBourne** [comment]
Duplicate of #564

**Comment by EmilyBourne** [comment]
This issue is fixed by PR #797 but we want to test whether marking it as a duplicate is enough for github to close it automatically when issue #564 is closed by the PR

**Comment by EmilyBourne** [comment]
Closed with duplicate issue #564
